src/process.rs

```diff
@@ -0,0 +1,340 @@
+                syncthing_proc.started_by_app = false; // External process
+
+                log::info!(
+                    "Detected {} Syncthing process with PID: {}",
+                    if external_only {
+                        "external"
+                    } else {
+                        "existing"
+                    },
+                    pid
+                );
+
+                return Ok(Some(syncthing_proc));
+            }
+        }
+        Ok(None)
+    }
+
+    /// Finds all Syncthing processes currently running on the system.
+    fn find_all_syncthing_processes(&mut self) -> Vec<u32> {
+        self.system.refresh_processes(ProcessesToUpdate::All, false);
+
+        // Get the executable name to search for
+        let exe_name = Path::new(&self.syncthing_path)
+            .file_name()
+            .and_then(|name| name.to_str())
+            .unwrap_or("syncthing");
+
+        // Remove .exe extension if present for cross-platform compatibility
+        let exe_name = exe_name.strip_suffix(".exe").unwrap_or(exe_name);
+
+        let mut pids = Vec::new();
+        for (pid, process) in self.system.processes() {
+            let process_name = process.name().to_string_lossy();
+            let process_name = process_name.strip_suffix(".exe").unwrap_or(&process_name);
+
+            if process_name.eq_ignore_ascii_case(exe_name) {
+                pids.push(pid.as_u32());
+            }
+        }
+
+        log::debug!("Found {} Syncthing processes: {:?}", pids.len(), pids);
+        pids
+    }
+
+    /// Updates the list of tracked Syncthing processes.
+    fn update_tracked_processes(&mut self) {
+        if self.started_by_app {
+            self.tracked_pids = self.find_all_syncthing_processes();
+        }
+    }    /// Starts a new Syncthing process.
+    pub fn start(&mut self, args: &[String]) -> io::Result<()> {
+        if self.child.is_some() {
+            return Err(io::Error::new(
+                io::ErrorKind::AlreadyExists,
+                "Process is already running",
+            ));
+        }
+
+        log::info!("Starting Syncthing process: {}", self.syncthing_path);
+
+        let mut command = Command::new(&self.syncthing_path);
+        command.args(args);
+        command.stdout(Stdio::null());
+        command.stderr(Stdio::null());
+        
+        // On Windows, prevent console window from appearing
+        #[cfg(windows)]
+        {
+            use std::os::windows::process::CommandExt;
+            command.creation_flags(0x08000000); // CREATE_NO_WINDOW
+        }
+        
+        let child = command.spawn()?;
+
+        self.pid = Some(child.id());
+        log::info!("Syncthing process started with PID: {}", child.id());
+
+        self.child = Some(child);
+        self.started_by_app = true;
+
+        // Give the process a moment to potentially spawn children
+        std::thread::sleep(std::time::Duration::from_millis(500));
```

Comment:
Is there some better way to detect when the child process has been actually spawned?